src/audio/drc/drc_math_lib/asin/c_fixpt/drc_asin_fixpt.c

```diff
@@ -0,0 +1,231 @@
+  }
+
+  return y;
+}
+
+ /*
+ *-----------------------------------------------------------------------------
+ *
+ * Prototype       :static scalar_cordic_asin
+ * Description     :Compute the inverse Sine using a CORDIC implementation.
+ * Arguments       :int8_T sinValue[2-wl,0,fl],int16_T nIters,const int8_T LUT[6][8-wl,5-fl]
+ * Returns         :int8_T
+ * Global Variables:None
+ *-----------------------------------------------------------------------------
+ */
+static int8_T scalar_cordic_asin(int8_T sinValue, int16_T nIters, const int8_T
+  LUT[6])
+{
+  int8_T z;
+  int8_T b_sinValue;
+  int8_T x;
+  int8_T y;
+  int32_T i;
+  int32_T b_i;
+  int16_T j;
+  int16_T k;
+  int8_T xShift;
+  int8_T xDShift;
+  int8_T yShift;
+  int8_T yDShift;
+  b_sinValue = (int8_T)(sinValue * 32);
+  if (b_sinValue > 22)
+  {
+    x = 0;
+    y = 64;
+    z = 50;
+  }
+  else
+  {
+    x = 64;
+    y = 0;
+    z = 0;
+  }
+
+  b_sinValue = (int8_T)(b_sinValue * 2);
+  i = ((int32_T)nIters) - 1;
+  if (i < -32768)
+  {
+    i = -32768;
+  }
+
+  for (b_i = 0; b_i < i; b_i++)	        /* Initialize temporary variables to be used in the CORDIC Iterations */
+
+  {
+    j = (int16_T)((b_i + 1) * 2);
+    if (j >= 7)
+    {
+      j = 7;
+    }
+
+    if (b_i < 7)
+    {
+      k = (int16_T)b_i;
+    }
+    else
+    {
+      k = 7;
+    }
+
+    xShift = (int8_T)asr_s32((int32_T)x, (uint32_T)k);
+    xDShift = (int8_T)asr_s32((int32_T)x, (uint32_T)j);
+    yShift = (int8_T)asr_s32((int32_T)y, (uint32_T)k);
+    yDShift = (int8_T)asr_s32((int32_T)y, (uint32_T)j);
+    if (y == b_sinValue)
+    /* Do nothing if x currently equals the target value. Please
+    * note that this is ONLY allowed for double rotations
+    * algorithms, as it is equivalent to rotating by the same angle
+    * in opposite directions sequentially. Note that it accounts
+    * for the scaling effect of CORDIC Pseudo-rotations as well
+    */
+    {
+      x += xDShift;
+      y += yDShift;
+    }
+    else if (((y >= b_sinValue) && (x >= 0)) || ((y < b_sinValue) && (x < 0)))
+    {
+      x = (int8_T)(((int8_T)(x - xDShift)) + yShift);
+      y = (int8_T)(((int8_T)(y - yDShift)) - xShift);
+      z -= LUT[b_i];
+    }
+    else
+    {
+      x = (int8_T)(((int8_T)(x - xDShift)) - yShift);
+      y = (int8_T)(((int8_T)(y - yDShift)) + xShift);
+      z += LUT[b_i];
+    }
+
+    b_sinValue += (int8_T)asr_s32((int32_T)b_sinValue, (uint32_T)j);
+  }
+
+  if (z < 0)
+  {
+    z = (int8_T)(-z);	/* Invert the sign, if necessary */
+  }
+
+  return z;
+}
+
+ /*^^^
+ *-----------------------------------------------------------------------------
+ *
+ * Prototype        :void drc_asin_fixpt
+ * Description      :calculate fixed point inverse sine of x based on a CORDIC approximation
+ * Input Arguments  :const int8_T x[2] Signedness:1, Signed,WordLength: 2,FractionLength: 0
+ * Output Arguments :int32_T y[2] Q30.32   [QFL.WL]
```

Comment:
The convention of presentation in SOF library should be Q(WL-FL).FL as I've seen elsewhere?
For example, Q30.32 here should be corrected to Q2.30
@lgirdwood please correct me if I'm wrong

Reply:
OK - I will change